__tests__/params.spec.ts

```diff
@@ -2,27 +2,79 @@ import { mount } from '@vue/test-utils'
 import { watch } from 'vue'
 import { getRouter } from '../src'
 
-describe('setParams', () => {
-  it('sets current route params', async () => {
-    const router = getRouter()
-    router.setParams({ userId: 12 })
-    const wrapper = mount({
-      template: `<p>{{ $route.params.userId }}</p>`,
+describe('Params utils', () => {
+  describe('`setParams`', () => {
+    it('sets current route params', async () => {
+      const router = getRouter()
+      router.setParams({ userId: 12 })
+      const wrapper = mount({
+        template: `<p>{{ $route.params.userId }}</p>`,
+      })
+      const spy = jest.fn()
+
+      watch(wrapper.vm.$route, spy)
+
+      expect(wrapper.vm.$route.params).toEqual({ userId: '12' })
+      expect(wrapper.text()).toBe('12')
+      expect(spy).toHaveBeenCalledTimes(0)
+
+      await router.setParams({ userId: 12 })
+      expect(spy).toHaveBeenCalledTimes(1)
+      expect(wrapper.text()).toBe('12')
+
+      await router.setParams({ userId: 2 })
+      expect(spy).toHaveBeenCalledTimes(2)
+      expect(wrapper.text()).toBe('2')
+    })
+  })
+
+  describe('`setQuery`', () => {
+    it('sets current route query', async () => {
+      const router = getRouter()
+      router.setQuery({ page: 2 })
+      const wrapper = mount({
+        template: `<p>{{ $route.query.page }}</p>`,
+      })
+      const spy = jest.fn()
+
+      watch(wrapper.vm.$route, spy)
+
+      expect(wrapper.vm.$route.query).toEqual({ page: '2' })
+      expect(wrapper.text()).toBe('2')
+      expect(spy).toHaveBeenCalledTimes(0)
+
+      await router.setQuery({ page: 2 })
+      expect(spy).toHaveBeenCalledTimes(1)
```

Comment:
I'm realizing this one should be 0 in all cases to be consistent with vue router (duplicated navigations) but I don't think I have exposed the logic for it yet so let's keep it this way for the moment